lib/config-generator.js

```diff
@@ -318,13 +318,13 @@ class ConfigGenerator {
         }
 
         if (this.webpackConfig.useEslintLoader) {
-            rules.push({
+            rules.push(applyOptionsCallback(this.webpackConfig.loaderConfigurationCallbacks['eslint'], {
```

Comment:
Hum why not.
It means that configuration rules (`{ test: ... }`) will be stored in `WebpackConfig#loaderConfigurationCallbacks`, or am I missing something?

Reply:
Nop, what I meant is that, instead of having:

```js
rules.push(applyOptionsCallback(this.webpackConfig.loaderConfigurationCallbacks['eslint'], {
    test: /\.jsx?$/,
    loader: 'eslint-loader',
    exclude: /node_modules/,
    enforce: 'pre',
    options: eslintLoaderUtil.getOptions(this.webpackConfig)
}));
```

You'd have:

```js
rules.push(applyRuleCallback('eslint', {
    test: /\.jsx?$/,
    loader: 'eslint-loader',
    exclude: /node_modules/,
    enforce: 'pre',
    options: eslintLoaderUtil.getOptions(this.webpackConfig)
}));
```

And `applyRuleCallback` would only be calling `applyOptionsCallback` using the right callback.
Maybe that could also be useful if we want to handle aliases (for instance `sass` and `scss`).

Reply:
Ah, yea that would be better.

Also I'm not sure how I will access the current `WebpackConfig` instance. 
Maybe pass it as the first parameter in my function and export it in another file, or create a private method in `ConfigGenerator`?

Reply:
Hm, it'll probably only be used inside of `buildRulesConfig`, so putting it beside it should be fine and allow access to `this.webpackConfig`.

Reply:
Ah yea, didn't see we have only one method to build every webpack rules. 

I will try to apply your review comments this afternoon.